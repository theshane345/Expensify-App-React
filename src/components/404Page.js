import {Link} from 'react-router-dom';
import React from 'react';

const NotFound = () =>(
    <div>
        404! NOT FOUND - <Link to="/">Go Home</Link>
    </div>
);

export default NotFound;

