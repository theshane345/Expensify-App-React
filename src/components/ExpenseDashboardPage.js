import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpenseSummary from './ExpensesSummary';

const Dashboard = () =>(
    <div>
        <ExpenseSummary/>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default Dashboard;