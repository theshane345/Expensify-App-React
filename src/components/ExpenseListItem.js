//Export a stateless functional componeent
//description, amount, 
import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({id, description, amount , createdAt}) =>(
    <div>
        <Link to= {`/edit/${id}`}>
            <h3>The expense {description}</h3> 
        </Link>
        <p>is of total amount {amount} created at {createdAt}</p>
        
    </div>
);

export default ExpenseListItem;
