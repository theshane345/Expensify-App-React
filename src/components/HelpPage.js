import React from 'react';

const Help= () =>(
    <div>
        <p>This is my help page you nerd</p>
    </div>
);

export default Help;