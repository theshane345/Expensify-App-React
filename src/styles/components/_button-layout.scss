.button-layout{
    background:$light-blue;
    border: none;
    color: white;
    display: inline-block;
    text-decoration: none;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    font-size: $font-size-l;
}

.button-layout--link{
    background: transparent;
    
}

.button-layout--secondary{
    @extend .button-layout;
    background: #888;
}

