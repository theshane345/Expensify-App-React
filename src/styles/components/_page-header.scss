.page-header{
    background: $off_white;
    padding: $l-size 0;
    margin-bottom: $l-size;
}

.page-header__actions{
    margin-top:$m-size;
}

.page-header__title{
    margin:0;
    font-weight: 300;
    span{
        font-weight: bold;
    }
}