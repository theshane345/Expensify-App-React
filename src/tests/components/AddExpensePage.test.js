import React from 'react';
import {shallow} from 'enzyme';
import{AddExpensePage} from '../../components/AddExpense';
import expenses from '../fixtures/expenses'

let addExpense,history,wrapper;

beforeEach(() => {
    addExpense = jest.fn();
    history = { push:jest.fn()};
    wrapper = shallow(<AddExpensePage addExpense={addExpense} history={history}/>);
});

test('should test that the add expense page renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should handle onSubmit', ()=>{
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1]);
    expect(history.push).toHaveBeenCalledWith('/');
    expect(addExpense).toHaveBeenCalledWith(expenses[1]);
});


