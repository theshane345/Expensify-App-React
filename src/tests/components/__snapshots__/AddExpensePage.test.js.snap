// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test that the add expense page renders correctly 1`] = `
<div>
  <h1>
    Add Expense
  </h1>
  <ExpenseForm
    onSubmit={[Function]}
  />
</div>
`;
